ena-submission/scripts/create_assembly.py

```diff
@@ -0,0 +1,527 @@
+                )
+                tries += 1
+            if number_rows_updated == 1:
+                logger.info(
+                    f"Assembly submission for accession {row["accession"]} succeeded and accession returned!"
+                )
+
+
+def assembly_table_handle_errors(
+    db_config,
+    config,
+    slack_config: SlackConfig,
+    time_threshold=15,
+    time_threshold_waiting=48,
+    slack_time_threshold=12,
+):
+    """
+    - time_threshold: (minutes)
+    - time_threshold_waiting: (hours)
+    - slack_time_threshold: (hours)
+    1. Find all entries in assembly_table in state HAS_ERRORS or SUBMITTING over time_threshold
+    2. If time since last slack_notification is over slack_time_threshold send notification
+    """
+    entries_with_errors = find_errors_in_db(
+        db_config, "assembly_table", time_threshold=time_threshold
+    )
+    if len(entries_with_errors) > 0:
+        error_msg = (
+            f"{config.backend_url}: ENA Submission pipeline found {len(entries_with_errors)} entries"
+            f" in assembly_table in status HAS_ERRORS or SUBMITTING for over {time_threshold}m"
+        )
+        send_slack_notification(
+            error_msg,
+            slack_config,
+            time=datetime.now(tz=pytz.utc),
+            time_threshold=slack_time_threshold,
+        )
+        # TODO: Query ENA to check if assembly has in fact been created
+        # If created update assembly_table
+        # If not retry 3 times, then raise for manual intervention
+    entries_waiting = find_waiting_in_db(
+        db_config, "assembly_table", time_threshold=time_threshold_waiting
+    )
+    if len(entries_waiting) > 0:
+        error_msg = (
+            f"ENA Submission pipeline found {len(entries_waiting)} entries in assembly_table in"
+            f" status WAITING for over {time_threshold_waiting}h"
+        )
+        send_slack_notification(
+            config, error_msg, time=datetime.now(tz=pytz.utc), time_threshold=slack_time_threshold
+        )
+
+
+@click.command()
+@click.option(
+    "--log-level",
+    default="INFO",
+    type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]),
+)
+@click.option(
+    "--config-file",
+    required=True,
+    type=click.Path(exists=True),
+)
+def create_assembly(log_level, config_file):
+    logger.setLevel(log_level)
+    logging.getLogger("requests").setLevel(logging.INFO)
+
+    with open(config_file) as file:
+        full_config = yaml.safe_load(file)
+        relevant_config = {key: full_config.get(key, []) for key in Config.__annotations__}
+        config = Config(**relevant_config)
+    logger.info(f"Config: {config}")
+
+    db_config = db_init(config.db_password, config.db_username, config.db_host)
+    slack_config = slack_conn_init(
+        slack_hook_default=config.slack_hook,
+        slack_token_default=config.slack_token,
+        slack_channel_id_default=config.slack_channel_id,
+    )
+
+    while True:
```

Comment:
Would it make sense to sleep a bit between the iterations?

Reply:
created a follow up issue: https://github.com/loculus-project/loculus/issues/2831